src/Microsoft.Framework.Runtime.Hosting/Json/JsonBuffer.cs

```diff
@@ -0,0 +1,301 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+
+namespace Microsoft.Framework.Runtime.Json
+{
+    internal class JsonBuffer
+    {
+        private readonly TextReader _reader;
+
+        private int _line = 1;
+        private int _column = 0;
+
+        public JsonBuffer(TextReader reader)
+        {
+            _reader = reader;
+        }
+
+        public JsonToken Read()
+        {
+            var result = new JsonToken();
+
+            int first;
+            while (true)
+            {
+                first = ReadNextChar();
+                result.Line = _line;
+                result.Column = _column;
+
+                if (first == -1)
+                {
+                    result.Type = JsonTokenType.EOL;
+                    return result;
+                }
+                else if (!IsWhitespace(first))
+                {
+                    break;
+                }
```

Comment:
No, the column is increased when a char is read (line 30). It doesn't matter if it s a space or not.